### ISSUE TITLE
question about path planning

### ISSUE THREAD

**Issue by Wayne-xixi** [ifaceyou123/point_cloud_fusion#3]
Hello,

When you used FCL-0.5 to catkin build path palnning, did you get this error?
Looking forward to your help!
'''CMakeFiles/path_planning_node.dir/src/path_planning.cpp.o: In function `fcl::Transform3f::transform(fcl::Vec3fX<fcl::details::Vec3Data<double> > const&) const':
path_planning.cpp:(.text._ZNK3fcl11Transform3f9transformERKNS_6Vec3fXINS_7details8Vec3DataIdEEEE[_ZNK3fcl11Transform3f9transformERKNS_6Vec3fXINS_7details8Vec3DataIdEEEE]+0x46): undefined reference to `fcl::Quaternion3f::transform(fcl::Vec3fX<fcl::details::Vec3Data<double> > const&) const'
CMakeFiles/path_planning_node.dir/src/path_planning.cpp.o: In function `fcl::CollisionGeometry::CollisionGeometry()':
path_planning.cpp:(.text._ZN3fcl17CollisionGeometryC2Ev[_ZN3fcl17CollisionGeometryC5Ev]+0x34): undefined reference to `fcl::AABB::AABB()'
CMakeFiles/path_planning_node.dir/src/path_planning.cpp.o: In function `fcl::CollisionObject::CollisionObject(std::shared_ptr<fcl::CollisionGeometry> const&)':
path_planning.cpp:(.text._ZN3fcl15CollisionObjectC2ERKSt10shared_ptrINS_17CollisionGeometryEE[_ZN3fcl15CollisionObjectC5ERKSt10shared_ptrINS_17CollisionGeometryEE]+0x59): undefined reference to `fcl::AABB::AABB()'
CMakeFiles/path_planning_node.dir/src/path_planning.cpp.o: In function `fcl::Box::Box(double, double, double)':
path_planning.cpp:(.text._ZN3fcl3BoxC2Eddd[_ZN3fcl3BoxC5Eddd]+0x29): undefined reference to `vtable for fcl::Box'
CMakeFiles/path_planning_node.dir/src/path_planning.cpp.o: In function `planner::isStateValid(ompl::base::State const*)':
path_planning.cpp:(.text._ZN7planner12isStateValidEPKN4ompl4base5StateE[_ZN7planner12isStateValidEPKN4ompl4base5StateE]+0x1e5): undefined reference to `fcl::collide(fcl::CollisionObject const*, fcl::CollisionObject const*, fcl::CollisionRequest const&, fcl::CollisionResult&)'
collect2: error: ld returned 1 exit status
make[2]: *** [/home/fapsros/catkin_ws/devel/.private/path_planning/lib/path_planning/path_planning_node] Error 1
make[1]: *** [CMakeFiles/path_planning_node.dir/all] Error 2
make: *** [all] Error 2
'''

**Comment by zhangzhenyuyu** [comment]
Have you solved the problem? I need your help，thank you

**Comment by Wayne-xixi** [comment]
> Have you solved the problem? I need your help，thank you

Yes, if u have met the same error above, its the problem of FCL version. I solved it by rewriting the code of FCL.

**Comment by zhangzhenyuyu** [comment]
@Wayne-xixi Thank you very much for your reply! Can you tell me some details about the solution process? I still can't solve this problem. Please!

**Comment by Wayne-xixi** [comment]
> @Wayne-xixi Thank you very much for your reply! Can you tell me some details about the solution process? I still can't solve this problem. Please!

Check your FCL Version, 0.5.0 or other？

**Comment by zhangzhenyuyu** [comment]
thank you very much! I have compiled successfully! Thank you very much! Now I have a new problem. I can't run this project without the introduction of the running process. I'd like to show you my current situation. Do you have any comments? Thank you very much!

```
rosrun path_planning path_planning
[rosrun] Couldn't find executable named path_planning below /home/zzy/path_planning-master
[rosrun] Found the following, but they're either not files,
[rosrun] or not executable:
[rosrun]   /home/zzy/path_planning-master/build/catkin_generated/stamps/path_planning
[rosrun]   /home/zzy/path_planning-master/build/devel/share/path_planning
[rosrun]   /home/zzy/path_planning-master/build/devel/lib/path_planning

```
I don't know how to run it

**Comment by Wayne-xixi** [comment]
> thank you very much! I have compiled successfully! Thank you very much! Now I have a new problem. I can't run this project without the introduction of the running process. I'd like to show you my current situation. Do you have any comments? Thank you very much!
> 
> ```
> rosrun path_planning path_planning
> [rosrun] Couldn't find executable named path_planning below /home/zzy/path_planning-master
> [rosrun] Found the following, but they're either not files,
> [rosrun] or not executable:
> [rosrun]   /home/zzy/path_planning-master/build/catkin_generated/stamps/path_planning
> [rosrun]   /home/zzy/path_planning-master/build/devel/share/path_planning
> [rosrun]   /home/zzy/path_planning-master/build/devel/lib/path_planning
> ```
> 
> I don't know how to run it

Firstly, you need to source your workspace after you built it. And then check the Pakagename in Cmake File.

**Comment by zhangzhenyuyu** [comment]
Thank you very much! I have run successfully, but I run every command without displaying the result. Do you know what this is?
```
rosrun path_planning 
goal_transform        path_planning_node    traj_gen
old_path_planning     scale                 transform_pointcloud
```

**Comment by Wayne-xixi** [comment]
> Thank you very much! I have run successfully, but I run every command without displaying the result. Do you know what this is?
> 
> ```
> rosrun path_planning 
> goal_transform        path_planning_node    traj_gen
> old_path_planning     scale                 transform_pointcloud
> ```

Can you show your terminal screen?

**Comment by zhangzhenyuyu** [comment]
OK，like this。There's nothing to show next（好的，像这样，下面就没有任何显示了）
![11](https://user-images.githubusercontent.com/54766833/82818942-1c19e000-9ed2-11ea-96ac-c75ce315d250.png)

**Comment by Wayne-xixi** [comment]
> OK，like this。There's nothing to show next（好的，像这样，下面就没有任何显示了）
> ![11](https://user-images.githubusercontent.com/54766833/82818942-1c19e000-9ed2-11ea-96ac-c75ce315d250.png)

GoalPointTransformer.cpp allow the users to select on goal point on the map through rviz
path_planning.cpp publishes trajectory_msgs::MultiDOFJointTrajectory "waypoint"
"waypoint" is taken as an input by trajectory_generation in order to produce a new trajectory_msg which is smoothed with jerk constraints
The scaling node is used to calculate the scale required for converting the monocular slam estimates to metric units.
The transform node is used to transform point cloud from camera axis convention to ROS convention.
The old_path_planner code is there only for reference since I have modified it in the path_planning node for replanning which is currently not working properly.

**Comment by Wayne-xixi** [comment]
> > OK，like this。There's nothing to show next（好的，像这样，下面就没有任何显示了）
> > ![11](https://user-images.githubusercontent.com/54766833/82818942-1c19e000-9ed2-11ea-96ac-c75ce315d250.png)
> 
> GoalPointTransformer.cpp allow the users to select on goal point on the map through rviz
> path_planning.cpp publishes trajectory_msgs::MultiDOFJointTrajectory "waypoint"
> "waypoint" is taken as an input by trajectory_generation in order to produce a new trajectory_msg which is smoothed with jerk constraints
> The scaling node is used to calculate the scale required for converting the monocular slam estimates to metric units.
> The transform node is used to transform point cloud from camera axis convention to ROS convention.
> The old_path_planner code is there only for reference since I have modified it in the path_planning node for replanning which is currently not working properly.

如果你还有问题可以加我wechat：waynexixi

**Comment by zhangzhenyuyu** [comment]
Thank you very much, I have added you as a friend, please agree
